interface/src/app/analyze/analysis/analysis.js

```diff
@@ -95,7 +98,11 @@ AnnotationType, SampleBin) {
       }, {
         'name': 'samples',
         'type': 'ordinal',
-        'domain': {'data': 'activity_normalized', 'field': 'sample'},
+        'domain': {
+          'data': 'activity_normalized',
+          'field': 'sample_order._id',
```

Comment:
Exactly. Vega can use a list of matching objects or just a plain list as a data source. The way it uses a plain list is by converting it to a list of objects. In either case, Vega captures the original list order by adding an `_id` field on to each object. For more details, see: https://github.com/vega/vega/wiki/Data